rules/aip0123/resource_pattern.go

```diff
@@ -18,40 +18,46 @@ import (
 	"fmt"
 	"strings"
 
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
 	"github.com/googleapis/api-linter/lint"
 	"github.com/googleapis/api-linter/locations"
 	"github.com/googleapis/api-linter/rules/internal/utils"
 	"github.com/jhump/protoreflect/desc"
+	"google.golang.org/genproto/googleapis/api/annotations"
 )
 
 var resourcePattern = &lint.MessageRule{
 	Name:   lint.NewRuleName(123, "resource-pattern"),
 	OnlyIf: hasResourceAnnotation,
 	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
 		resource := utils.GetResource(m)
-		// Are any patterns declared at all? If not, complain.
-		if len(resource.GetPattern()) == 0 {
+		return lintResourcePattern(resource, m, locations.MessageResource(m))
+	},
+}
+
+func lintResourcePattern(resource *annotations.ResourceDescriptor, desc desc.Descriptor, loc *dpb.SourceCodeInfo_Location) []lint.Problem {
```

Comment:
Should we also check that not only do the resource patterns not contain `_` but also that the pluralized words were translated to singular, and correctly? We could use, https://github.com/gertd/go-pluralize? (Obviously, we'd have to convert camel to snake and vice versa, but could be done~)

Also, we check that the segments **do not** contain `_` but we do not check that if there is a capital letter in the fixed segment, indicating that there are multiple words because it is camel case, that the static segment **does** contain an `_`.

Reply:
> but also that the pluralized words were translated to singular, and correctly

Could you give an example? And could we do it in a separate PR/issue?

Reply:
> we do not check that if there is a capital letter in the fixed segment, indicating that there are multiple words because it is camel case, that the static segment does contain an _.

I'm sorry you lost me 😅

Reply:
Yes, we can do it in another PR. Let's take it up offline.